src/Security/UserClassBuilder.php

```diff
@@ -97,10 +97,18 @@ private function addGetUsername(ClassSourceManipulator $manipulator, UserClassCo
             );
         }
 
+        $getterIdentifierName = 'getUsername';
+
+        $reflectedInterface = new \ReflectionClass(UserInterface::class);
+
+        if (false !== stripos($reflectedInterface->getDocComment(), '@method string getUserIdentifier()')) {
+            $getterIdentifierName = 'getUserIdentifier';
+        }
```

Comment:
What if we just look for the presence of this concrete method: https://github.com/symfony/symfony/blob/527b885fc68da965ada884e484d043f79f703f45/src/Symfony/Component/Security/Core/User/User.php#L96-L99 - looking for `@method` is a no go because it would move to be a real method in 6.0

Reply:
ahh yes i like that a lot better.. saves us from having to worry about adding another conditional for 6+

Reply:
done